java/sage/miniclient/MiniClient.java

```diff
@@ -154,12 +163,34 @@ public void write(int b) {}
 
       public synchronized void println(String s)
       {
-        System.err.println(df() + " " + s);
+        if (myProperties.getProperty(redirectLogProp, "false").equalsIgnoreCase("true")){
+          if (printWriter == null) {
+
+              try { printWriter = new PrintWriter(redirectLog); } catch (Exception e){}
+          }
+          printWriter.println(df() + " " + s);
+        }else{
+          System.err.println(df() + " " + s);
+        }
       }
     };
 
-    System.setOut(redir);
-    //System.setErr(redir);
+    if (myProperties.getProperty(redirectLogProp, "false").equalsIgnoreCase("true")){
+        try {
+           System.setOut(new PrintStream(redirectLog));
+            } catch (Exception e) {
+            e.printStackTrace();
+        }
+        try {
```

Comment:
Don't create 2 separate PrintStream objects here because they will end up overwriting each other (or actually failing because two things are trying to open the same file for writing). Create one, and set it for both output streams. I'd also recommend constructing it like:  new PrintStream(new BufferedOuptutStream(new FileOutputStream)), true); We want to have automatic line flushing or you can lose important data when a crash occurs.

Reply:
corrected